solutions/devsprint-matheus-reis-7/FinanceApp/Service/FinanceService.swift

```diff
@@ -7,9 +7,31 @@
 
 import Foundation
 import Combine
+import RxSwift
 
-class FinanceService {
+enum FinanceServiceEndpoints {
+    case activityDetails
 
+    var urlString: String {
+        switch self {
+        case .activityDetails:
+            return "https://raw.githubusercontent.com/devpass-tech/challenge-viewcode-finance/10ce6c2e9c88199ad8c4e721212099f55b26dfbb/api/activity_details_endpoint.json"
+        }
+    }
+}
+
+final class FinanceService {
+
+    private let urlSession: URLSession
+    private let jsonDecoder: JSONDecoder
+
+    init(urlSession: URLSession = URLSession.shared,
+         jsonDecoder: JSONDecoder = JSONDecoder()) {
+        self.urlSession = urlSession
+        self.jsonDecoder = jsonDecoder
+    }
+
+    // MARK: - fetchHomeData
     func fetchHomeData() -> AnyPublisher<HomeData?, Error> {
```

Comment:
sanity-check: Are you using `Combine` and `RxSwift` on same project?

Reply:
Is an implementation of another developer, for study puporses. Out tech lead made the framework choice optional for us.

Reply:
Sounds good!